toolchain/check/facet_type.cpp

```diff
@@ -671,18 +650,18 @@ auto ResolveFacetTypeRewriteConstraints(
       continue;
     }
 
-    cache.SetBeingRewritten(context, *lhs_access);
+    rewrite_values.SetBeingRewritten(context, *lhs_access);
 
     auto replace_witness_callbacks =
-        SubstImplWitnessAccessCallbacks(&context, loc_id, rewrites, &cache);
+        SubstImplWitnessAccessCallbacks(&context, loc_id, &rewrite_values);
     auto subst_inst_id =
         SubstInst(context, constraint.rhs_id, replace_witness_callbacks);
     constraint.rhs_id = subst_inst_id;
     if (constraint.rhs_id == SemIR::ErrorInst::InstId) {
       return false;
     }
 
-    cache.InsertRewritten(context, *lhs_access, subst_inst_id);
+    rewrite_values.SetFullyRewritten(context, *lhs_access, subst_inst_id);
```

Comment:
It strikes me that there are no shape-changing operations on `rewrite_values` within this loop, given that `SubstImplWitnessAccessCallbacks` no longer calls insert, so we could avoid doing this lookup and reuse the result from the `SetBeingRewritten` lookup too :)